app/models/user.rb

```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  before_save{self.email = email.downcase}
+  validates :name, presence: true, length: {maximum: 50}
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  validates :email, presence: true, length: {maximum: 255},
+    format: {with: VALID_EMAIL_REGEX}, uniqueness: {case_sensitive: false}
+  has_secure_password
```

Comment:
`has_secure_password` cung cấp những gì?

Reply:
Lưu thuộc tính password_digest an toàn vào cơ sở dữ liệu
2 thuộc tính password và password_confirmation xác nhận 2 thuộc tính này hợp lệ và trùng khớp
Một authenticate method trả về user khi mật khẩu đó là đúng (và sai khi ngược lại)